### PR TITLE
#added preference for column type visibility

### PR DESCRIPTION
**Pull request by DannyJJK**
## Changes:
- Add preference for column type visiblity on content view (Preferences.xib is modified but the alignment needs fixing)

## Closes following issues:
- Closes: #1614

## Tested:
- Processors:
  - [x] Intel
  - [ ] Apple Silicon
- macOS Versions:
  - [ ] 10.13.x (High Sierra)
  - [ ] 10.14.x (Mojave)
  - [ ] 10.15.x (Catalina)
  - [ ] 11.x (Big Sur)
  - [ ] 12.x (Monterey)
  - [x] 13.x (Ventura)
- Localizations:
  - [ ] English
  - [x] Spanish
  - [ ] Other (please specify)
- Xcode Version: 14.1.0

## Additional notes:
I haven't worked with Objective-C or Xcode before so this is a bit of a rough implementation. See the following points:
- **I have tried to update Preferences.xib but the alignment is currently wrong. This will need fixing.**
- The table content won't automatically update if it's currently open when the preference is changed.

### CODE REVIEW

**Review by Kaspik — COMMENTED** [review]
Can you post a screenshot so we can see what's wrong without building it? :)

**Comment by DannyJJK** [comment]
image

Sorry @Kaspik I thought you would be able to see it inside Xcode. Here's the screenshot. I'm not used to the Xcode Interface Builder, I'm used to building interfaces with code rather than a WYSIWYG builder.

**Review by Jason-Morcos — CHANGES_REQUESTED** [review]
> image
> 
Looks like the checkbox is overlapping the line?
That means the constraint on the bottom of Save Query History Individually needs to be removed and a new constraint added to the bottom of your new checkbox.

Basically the auto layout constraints need to be adjusted

File: Source/Controllers/MainViewControllers/TableContent/SPTableContent.m [hunk 1]
```diff
@@ -527,7 +527,13 @@ - (void)setTableDetails:(NSDictionary *)tableDetails {
 
 		// Set up the column
 		theCol = [[NSTableColumn alloc] initWithIdentifier:[columnDefinition objectForKey:@"datacolumnindex"]];
-		[[theCol headerCell] setAttributedStringValue:[columnDefinition tableContentColumnHeaderAttributedString]];
+
+        if ([prefs boolForKey:SPDisplayTableViewColumnTypes]) {
+            [[theCol headerCell] setAttributedStringValue:[columnDefinition tableContentColumnHeaderAttributedString]];
+        } else {
+            [[theCol headerCell] setStringValue:[columnDefinition objectForKey:@"name"]];
+        }
```
Comment (Jason-Morcos) [inline Source/Controllers/MainViewControllers/TableContent/SPTableContent.m:535]: There probably also needs to be a key value observer set up to reload the table content when the option is changed

**Comment by DannyJJK** [comment]
Thank you @Jason-Morcos. I'm new to this so I'm still figuring the auto layout out. I will see if I can get the table layout to reload as well. I'll probably be able to look at this again tomorrow, I'm based in the UK. Thank you both for your input.

**Comment by Jason-Morcos** [comment]
Happy to help! We love having folks contribute to Ace!

**Comment by Jason-Morcos** [comment]
@DannyJJK Any updates here? Need anything from us? :)

**Comment by DannyJJK** [comment]
Sorry I hadn't got round to looking at this. I'll try to have a look either today or tomorrow

File: Source/Controllers/MainViewControllers/TableContent/SPTableContent.m (hunk 1 above)
Comment (DannyJJK) [reply Source/Controllers/MainViewControllers/TableContent/SPTableContent.m:535]: > > image
> 
> Looks like the checkbox is overlapping the line? That means the constraint on the bottom of Save Query History Individually needs to be removed and a new constraint added to the bottom of your new checkbox.
> 
> Basically the auto layout constraints need to be adjusted

@Jason-Morcos  I'm struggling to get the auto-layout to work so I think I'll need a hand with this. I have a feeling it's simple when you know how, but unfortuantely I've got no experience with Xcode or auto-layout/interface builder.

Comment (DannyJJK) [reply Source/Controllers/MainViewControllers/TableContent/SPTableContent.m:536]: > There probably also needs to be a key value observer set up to reload the table content when the option is changed

@Jason-Morcos I've tried adding an observer in my latest commit but it doesn't appear to be working :(. Can you see what's wrong with it? Is it possible that `[tableContentView reloadData];` only reloads the actual data of the table and not the headers?

**Comment by Kaspik** [comment]
I'll merge this to the local branch so I can finish the UI and I'll take a look at the observer. Thanks for your PR!

**Pull request merged**